wt-status.c

```diff
@@ -651,6 +651,14 @@ static void wt_status_collect_changes_index(struct wt_status *s)
 	rev.diffopt.detect_rename = s->detect_rename >= 0 ? s->detect_rename : rev.diffopt.detect_rename;
 	rev.diffopt.rename_limit = s->rename_limit >= 0 ? s->rename_limit : rev.diffopt.rename_limit;
 	rev.diffopt.rename_score = s->rename_score >= 0 ? s->rename_score : rev.diffopt.rename_score;
+
+	/*
+	 * The `recursive` flag must be set to properly perform a diff on sparse
+	 * directory entries.
+	 */
+	if (s->repo->index->sparse_index)
+		rev.diffopt.flags.recursive = 1;
```

Comment:
I wonder if there is any negative to setting `recursive = 1` in all cases. My guess is that it will be correct, but possibly slower? It would be great if it wasn't actually any slower and we don't need the special case.

Reply:
I'll check this (and add the performance numbers to the commit message) once the functional issues in this series are ironed out. For now, I'll leave this conversation open so I don't forget about it. 🙂

Reply:
It turns out that, even on full-index `git status` the `recursive` flag didn't affect performance in any noticeable way! Probably because a normal `git status` diff isn't normally running a diff on a directory anyway. Sparse index check removed + performance results added in commit message.